### PR TITLE
Big cAVS tooling update

### PR DESCRIPTION
**Pull request by andyross**
The confusion about different scripting environments and how to do testing using tools like twister was getting a little out of control.  This steals the cavstool.py patch out of the IPC PR and updates the older twister integration (now named "cavstwist.sh") in the same directory. 

And it kills all the older scripts in the process.

### CODE REVIEW

File: soc/xtensa/intel_adsp/tools/cavstool.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        (wlen, start1, end, seq1) = win_hdr()
+        if start1 == start and seq1 == seq:
+            return (seq, result.decode("utf-8"))
+
+async def main():
+    global hda, sd, dsp, hda_ostream_id, cavs15
+    (hda, sd, dsp, hda_ostream_id, cavs15) = map_regs()
+    log.info(f"Detected cAVS {'1.5' if cavs15 else '1.8+'} hardware")
+
+    load_firmware(sys.argv[1])
```
Comment (teburd) [inline soc/xtensa/intel_adsp/tools/cavstool.py:288]: Allow for logging only perhaps, though perilous....

```suggestion
    if len(sys.argv) > 1:
        load_firmware(sys.argv[1])
```

File: soc/xtensa/intel_adsp/tools/cavstwist.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+# SPDX-License-Identifier: Apache-2.0
+set -e
+
+# Twister integration tool for a remote ssh-accessible cAVS audio DSP
+# host (e.g. an Up Squared board running Linux). Can be used as the
+# hook for both --device-serial-pty and --west-flash, for example:
+#
+#  export CAVS_HOST=tgl2
+#  export CAVS_KEY=$HOME/otc_private_key_3k.pem
+#  export CAVS_RIMAGE=$HOME/rimage
```
Comment (teburd) [inline soc/xtensa/intel_adsp/tools/cavstwist.sh:12]: I'd add a note/comment here that this is the path to a directory containing rimage and the config directory, this bit me and I had to look harder at what was going on.

**Review by teburd — CHANGES_REQUESTED** [review]
much much nicer than my own cobbled scripts from the cavsload.sh that existed before, seems to work quite well for me

Really +1 with some minor nits to help the next person that does what I did

**Review by teburd — APPROVED** [review]
I'm able to reliably do twister runs without fail now with this

**Review by nashif — CHANGES_REQUESTED** [review]

File: boards/xtensa/intel_adsp_cavs15/tools/adsplog.py [hunk 3]
```diff
@@ -1,104 +0,0 @@
-#!/usr/bin/python3
```
[the remark this answers is not in the document]
Comment (andyross) [reply boards/xtensa/intel_adsp_cavs15/tools/adsplog.py:1]: cavstool -l, see last patch

[the remark this answers is not in the document]
Comment (nashif) [reply boards/xtensa/intel_adsp_cavs15/tools/adsplog.py:1]: oh sorry, those were comments I wrote earlier and did not submit....

File: soc/xtensa/intel_adsp/tools/cavstwist.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+if [ "$DO_SIGN" = "1" ]; then
+    # Wait for the last flasher to accept the old file
+    while [ -e $IMAGE ]; do
+	sleep 0.1
+    done
+
+    ELF=$BLDDIR/zephyr/zephyr.elf.mod
+    BOOT=$BLDDIR/zephyr/bootloader.elf.mod
+    (cd $BLDDIR; west sign --tool-data=$CAVS_RIMAGE/config -t rimage -- -k $CAVS_KEY) >/dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (andyross) [reply soc/xtensa/intel_adsp/tools/cavstwist.sh]: fixed

**Review by nashif — APPROVED** [review]

**Pull request merged**